libs/core/device-initialization/actions/initialize-mudita-pure.action.ts

```diff
@@ -36,72 +36,94 @@ export const initializeMuditaPure = createAsyncThunk<
 >(
   DeviceInitializationEvent.InitializeMuditaPure,
   async (_, { dispatch, getState, rejectWithValue }) => {
-    dispatch(
-      setDeviceInitializationStatus(DeviceInitializationStatus.Initializing)
-    )
-    await dispatch(loadSettings())
-
-    const unlockStatus = await dispatch(getUnlockStatus())
-    if (!unlockStatus.payload) {
-      return DeviceInitializationStatus.Initializing
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const dispatchWithDeviceCheck = async (action: any): Promise<any> => {
+      if (!isActiveDeviceAttachedSelector(getState())) {
+        dispatch(
+          setDeviceInitializationStatus(DeviceInitializationStatus.Aborted)
+        )
+        throw new AppError(DeviceInitializationError.ActiveDeviceNotAttached)
+      }
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+      return dispatch(action)
```

Comment:
please remove 'any' and eslint-disables

Reply:
and second, do we really need to check if device is attached every time we do some reading action from device? Sounds like overkill, but maybe there is some scenario for it.

Reply:
> please remove 'any' and eslint-disables

I understand the request to remove 'any' and eslint-disables. However, I'm finding it challenging to manage types for this helper in a straightforward manner. Therefore, I've opted for a compromise to avoid overengineering with types, especially since this helper is confined within the action context and doesn't extend beyond it."

Reply:
> and second, do we really need to check if device is attached every time we do some reading action from device? Sounds like overkill, but maybe there is some scenario for it.

I understand your perspective and have been pondering how to handle this differently myself. Perhaps I'm missing something or not seeing the full picture. Could we have a brief sync on this topic? If I explain the issue in detail, it might spark a new idea for me, or you might be able to offer some valuable advice.

Reply:
After sync review, refactored code to eliminate abstractions and use if statements before invoking async thunks
https://github.com/mudita/mudita-center/pull/1687/commits/f79796a8594fad0ab72bd15b2a3226b29abcaeaa